src/site/markdown/in_depth_walkthrough.md

```diff
@@ -0,0 +1,316 @@
+data that can encode fields with types: 
+
+- Booleans
+- Enumerated Type Values (Categorical Variables)
+- 32-bit and 64-bit integers
+- 32-bit and 64-bit floating point numbers
+- Strings
+- Vectorized inputs
+- Optional inputs
+
+Specifying the plugin is easy.  It just requires providing a column name to column type map.  The types come from
+[com.eharmony.aloha.semantics.compiled.plugin.csv.CsvTypes](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.plugin.csv.CsvTypes%24).
+
+```scala
+// Scala Code
+import com.eharmony.aloha.semantics.compiled.plugin.csv.CsvTypes.CsvType._
+
+val plugin = CompiledSemanticsCsvPlugin(
+  "height" -> IntType
+  "weight" -> FloatOptionType
+)
+```
+
+Notice that since [CompiledSemanticsCsvPlugin](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.plugin.csv.CompiledSemanticsCsvPlugin) extends 
+[CompiledSemanticsPlugin](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.CompiledSemanticsPlugin)\[[CsvLine](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.plugin.csv.CsvLine)\],
+the features produced by a [CompiledSemantics](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.CompiledSemantics) instance 
+parameterized by an [CompiledSemanticsCsvPlugin](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.plugin.csv.CompiledSemanticsCsvPlugin) operate 
+on instances of [CsvLine](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.plugin.csv.CsvLine).  It 
+is therefore necessary to produce instances of CsvLine to make use of this plugin.  This is pretty easy to do though.
+
+```scala
+val csvLines = CsvLines(
+  Map(                     // 0-based field indices
+    "height" -> 0, 
+    "weight" -> 1
+  ),
+  Map.empty,               // enums: Map[String, Enum] 
+  "\t",                    // fs (field separator) 
+  ",",                     // ifs (intra-field separator, for vectorized inputs)
+  (s: String) => s == "",  // missingData function 
+  false,                   // errorOnOptMissingField 
+  false                    // errorOnOptMissingEnum
+)
+
+val csvLinesWithDefaults = CsvLines(Map("height" -> 0, "weight" -> 1))
+
+require(csvLinesWithDefaults == csvLines)  // This requirement is satisfied.
+```
+
+Once in possession of a CsvLines instance, it's easy to generate CsvLine instances:  
+
+```scala
+val oneCsvLine: CsvLineImpl = csvLines("72, 175") // height = 72 inches, weight = 175 lbs.
+val manyCsvLinesUsingVarArgs: Vector[CsvLineImpl] = csvLines("72, 175", "70, 180", "64, 110")
+val csvLineIterator: Iterator[CsvLineImpl] = csvLines(Iterator("72, 175", "70, 180", "64, 110"))
+val csvLineVector: Seq[CsvLineImpl] = csvLines(Seq("72, 175", "70, 180", "64, 110"))
+val nonStrictVectorOfCsvLine = csvLines.nonStrict(Vector("72, 175", "70, 180", "64, 110"))
+```
+
+## Factory
+
+Factories are important to create models: 
```

Comment:
A few more sentences need to be added here.  At the very least a link to the wiki page about factory patterns.  Show an example with protobufs as well.  Also is this the appropriate place to show Java code for model instantiation as well?  What about code to create semantics?